### PR TITLE
Unify towards k8s.io/utils/clock - part 2

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #94738

```release-note
NONE
```

/kind cleanup
/priority important-longterm

### CODE REVIEW

File: staging/publishing/import-restrictions.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@
   allowedImports:
   - k8s.io/apimachinery
   - k8s.io/kube-openapi
+  - k8s.io/utils/clock
```
Comment (wojtek-t) [inline staging/publishing/import-restrictions.yaml:38]: Please double check this one :)

**Comment by wojtek-t** [comment]
/assign @lavalamp 

Please take a look when you find a moment - second part of mechanical changes (modulo one commit where I update import restrictions - please double check that). The third PR I'm planning to do after this one should finalize this migration.

File: staging/publishing/import-restrictions.yaml (hunk 1 above)
Comment (lavalamp) [reply staging/publishing/import-restrictions.yaml:38]: Seems fine!

**Pull request merged**